### PR TITLE
Added shared storage ops and migrate a TC

### PR DESCRIPTION
**Pull request by nik-redhat**
Migrated TC [test_shared_storage](https://github.com/gluster/glusto-tests/blob/master/tests/functional/glusterd/test_shared_storage.py)

Added `shared_storage_ops`

Fixes: #609 , and Fixes: #610 
Updates: #292 
Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

**Comment by nik-redhat** [comment]
Test run:
```
Table:
 test_shared_storage
 
+-------------+-------------+------------------+
| Volume Type | Test Result | Time taken (sec) |
+-------------+-------------+------------------+
|     dist    |     PASS    | 296.097496509552 |
+-------------+-------------+------------------+



Framework runtime : 306.606241941452
```

File: common/ops/gluster_ops/shared_storage_ops.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        try:
+            self.set_volume_options("all", option, node)
+
+        except Exception as error:
+            self.logger.error(f"Failed to disable shared storage: {error}")
+            return False
+
+        return True
+
+    def is_shared_volume_mounted(self, node: str) -> bool:
```
Comment (srijan-sivakumar) [inline common/ops/gluster_ops/shared_storage_ops.py:59]: Suggestion: We can take timeout as an optional argument.

File: common/ops/gluster_ops/shared_storage_ops.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                self.logger.info("Shared storage is mounted")
+                return True
+
+            sleep(2)
+            counter += 2
+
+        self.logger.info("Shared storage not mounted")
+        return False
+
+    def check_gluster_shared_volume(self, node: str,
```
Comment (srijan-sivakumar) [inline common/ops/gluster_ops/shared_storage_ops.py:85]: Suggestion: Timeout as optional argument.

File: common/ops/gluster_ops/shared_storage_ops.py (hunk 1 above)
Comment (nik-redhat) [reply common/ops/gluster_ops/shared_storage_ops.py]: Done

File: common/ops/gluster_ops/shared_storage_ops.py (hunk 2 above)
Comment (nik-redhat) [reply common/ops/gluster_ops/shared_storage_ops.py:86]: Done

**Review by srijan-sivakumar — APPROVED** [review]

**Pull request merged**